lib/board.rb

```diff
@@ -22,62 +24,36 @@ def validate_coordinate?(coordinate)
       key == coordinate
     end
   end
-
   def build_grid
+
     grid = {}
     ('A'..'D').each do |letter|
       (1..4).each do |num|
+        @letters << letter
+        @numbers << num
         coordinates = "#{letter}#{num}"
         grid[coordinates] = Cell.new(coordinates)
       end
     end
     grid
   end
 
-
   def place(ship,coordinates)
     coordinates.each do |coordinate|
     @cells[coordinate].place_ship(ship)
     end
   end
 
-  def render
-    number_string = []
-    key_numbers = @cells.keys.map do |key|
-      number_string.push(key[1])
-    end
-    number_header = " #{number_string.uniq.join(" ")} \n"
-
-    numbers = key_numbers.flatten.uniq
-
-    letter_header = ""
-    key_renders = ""
-    numbers.each do |number|
-      letter_header += "#{(number.to_i + 64).chr} . . . . \n"
-    end
-
-    #
-    # @cells.keys.each do |key|
-    #   key[0] == letter_header
-
-
-
-
-
-
-
-
-
-
-
- #      key_render = letter_header.map do |line|
- # @cells.keys.[0] == letter_header[0]
- #        letter_header += " #{key_render}"
- #
- #
- #      key_renders {@cells.keys.map{ |key| key[0].ord == letter_header[0].ord}}
-
-
-    " #{number_header}" + "#{letter_header}"
+  def render(reveal = false)
+    board = "  #{@numbers.uniq.join(" ")} \n"
+    letter_lines = ""
```

Comment:
This is how we use nested enumerables!

Pay attention to where we missed adding spaces